examples/java8/src/main/java/org/apache/beam/examples/Snippets.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+
+/**
+ * Code snippets used in webdocs.
+ */
+public class Snippets {
+
+  /* Helper function to format results in CoGroupByKeyTuple */
+  public static String formatCoGbkResults(
+      String name, Iterable<String> emails, Iterable<String> phones) {
+
+    List<String> emailsList = new ArrayList<>();
+    for (String elem : emails) {
+      emailsList.add("'" + elem + "'");
+    }
+    Collections.<String>sort(emailsList);
+    String emailsStr = "[" + String.join(", ", emailsList) + "]";
+
+    List<String> phonesList = new ArrayList<>();
+    for (String elem : phones) {
+      phonesList.add("'" + elem + "'");
+    }
+    Collections.<String>sort(phonesList);
+    String phonesStr = "[" + String.join(", ", phonesList) + "]";
+
+    return name + "; " + emailsStr + "; " + phonesStr;
+  }
+
+  public static PCollection<String> coGroupByKeyTuple(
+      Pipeline p,
+      TupleTag<String> emailsTag,
+      TupleTag<String> phonesTag,
+      List<KV<String, String>> emailList,
+      List<KV<String, String>> phoneList) {
+
+    // [START CoGroupByKeyTuple]
+    PCollection<KV<String, String>> emailsPColl = p.apply("create emails", Create.of(emailList));
```

Comment:
<!--new_thread; commit:b7cd18485fb9d3c89bc5177ad8ab8742faa39c06; resolved:0-->
Could these be provided (e.g. not via create, via a method call or related?)

Reply:
I'm not sure I understand what you mean by this. Do you mean passing in the emails and phones PCollections directly as arguments instead of creating them here from lists?

If so, I will also need to update the Python snippets to match for docs consistency:
[snippets.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets.py#L1140)
[snippets_test.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py#L696)

Reply:
Yeah. You can construct them from lists upstream, but I think the `Create` is unnecessary.